packages/sanity/src/tasks/src/tasks/components/form/tasksFormBuilder/TasksNotificationTarget.tsx

```diff
@@ -33,39 +33,45 @@ function TasksNotificationTargetInner(props: ObjectFieldProps<TaskDocument>) {
   const imageUrl = isImageSource(value?.media)
     ? imageBuilder.image(value.media).width(96).height(96).url()
     : null
+  const notificationTarget: TaskContext['notification'] = useMemo(() => {
+    const contextUrl = new URL(context?.notification?.url || '')
+    const currentUrl = new URL(`${window.location.origin}${basePath}/`)
 
-  useEffect(() => {
-    if (documentId && documentType && previewValuesLoading) {
-      // Wait until the preview values are loaded
-      return
-    }
+    const studioUrl =
+      contextUrl.hostname !== 'localhost' && currentUrl.hostname === 'localhost'
```

Comment:
Theoretically this can be `127.0.0.1`, `0.0.0.0`, `::1` or a local IP as well. Maybe you can import `isDev` from `sanity` instead and use that? Not quite the same, but might be good enough?